modules/apps/frontend-js/frontend-js-item-selector-web/src/main/resources/META-INF/resources/item_selector/DetachedItemSelectorModal.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {useModal} from '@clayui/modal';
+import React, {useEffect} from 'react';
+
+import {IItemSelectorModalProps} from './ItemSelectorModal';
+import ItemSelectorModalContent from './ItemSelectorModalContent';
+
+export type TDetachedItemSelectorModal<T> = Omit<
+	IItemSelectorModalProps<T>,
+	'observer' | 'onOpenChange' | 'open'
+>;
+
+function DetachedItemSelectorModal<T extends Record<string, any>>(
+	props: TDetachedItemSelectorModal<T>
+) {
+	const {observer, onOpenChange, open} = useModal();
+
+	useEffect(() => {
+		onOpenChange(true);
+	}, [onOpenChange]);
+
+	return (
+		<>
+			{open && (
+				<ItemSelectorModalContent
```

Comment:
Reusing everything apart from opening mechanism. No changes internal to `<ClayModal>`.

Reply:
@markocikos why is it necessary to create ItemSelectorModalContent? Couldn't you just use the ItemSelectorModal component here and default open to true?

Reply:
I can open it, but cannot close it. There is no way to control open state, as in current implementation it is [controlled prop with state outside of the component](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-item-selector-web/src/main/resources/META-INF/resources/item_selector/ItemSelector.tsx#L73-L105). I cannot use `useModal` in JS-only context in CKEditor plugin.

Reply:
You are very right! I pushed the simplification.